src/Test/Utilities/Portable/TestBase.cs

```diff
@@ -237,7 +237,10 @@ public static MetadataReference MscorlibRef
             {
                 if (s_mscorlibRef == null)
                 {
-                    s_mscorlibRef = AssemblyMetadata.CreateFromImage(TestResources.NetFX.v4_0_30319.mscorlib).GetReference(display: "mscorlib.v4_0_30319.dll");
+                    Interlocked.CompareExchange(
+                        ref s_mscorlibRef,
+                        AssemblyMetadata.CreateFromImage(TestResources.NetFX.v4_0_30319.mscorlib).GetReference(display: "mscorlib.v4_0_30319.dll"),
+                        null);
```

Comment:
As far as I can tell, only mscorlib is commonly depended on for symbol equality, which is the only meaningful difference in testing. I think we should be safe just using cmpxchg for mscorlib.